### PR TITLE
refactor(store): use `Object.is` as default equality check

### PR DESCRIPTION
**Pull request by arturovt**
In this commit, we default to using `Object.is` as the equality check function (this function is also used by other frameworks). `Object.is` supports checking for `NaN` values internally. We have also removed the reset function, as it was unused. If needed in the future, it can be re-added.

### CODE REVIEW

File: packages/store/internals/src/memoize.ts [hunk 1]
```diff
@@ -59,10 +43,5 @@ export function ɵmemoize<T extends (...args: any[]) => any>(
     lastArgs = arguments;
     return lastResult;
   }
-  (<any>memoized).reset = function () {
-    // The hidden (for now) ability to reset the memoization
-    lastArgs = null;
-    lastResult = null;
-  };
```
Comment (markwhitfeld) [inline packages/store/internals/src/memoize.ts:66]: The removal of this function is unrelated to this PR.
And besides, what is the motivation to remove it?

PS. The PR also requires an update to the changelog

**Review by markwhitfeld — COMMENTED** [review]
Great find!!!

File: packages/store/internals/src/memoize.ts (hunk 1 above)
Comment (arturovt) [reply packages/store/internals/src/memoize.ts:66]: I can remove it in another PR, but that part should not be included in the code unless it’s ever used. Extra work is bad, so unless there’s a valid argument for including that extra code, it shouldn’t be there.
